File: Drink Mixer 9000/app/src/main/java/greenbrier/drinkmixer3000/MainActivity.java
package greenbrier.drinkmixer3000;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.AsyncTask;
import android.os.Bundle;
import android.os.Debug;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.concurrent.locks.ReentrantLock;

/**
 * Created by McGiv on 8/31/2017.
 */

public class MainActivity extends AppCompatActivity
{
    public static final String DRINKS_PARCELABLE_NAME = "MIX_N_DRINKS";

    private Network net;

    private static int pin;
    private static ReentrantLock pinLock;
    private static Machine machine;

    private ArrayList<Drink> drinks;
    private ArrayList<Mix> mixes;

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        Log.d("instances", "onCreate Called");

        setContentView(R.layout.activity_main);

        drinks = new ArrayList<Drink>();
        mixes = new ArrayList<Mix>();

        pinLock = new ReentrantLock();

        machine = new Machine();

        if(isNetworkConnected())
        {
            net = new Network();
            net.execute(new WorkOrder(Network.INIT, mixes, drinks, getMachineDrinks()));
        }
        else
        {
            Log.d("Network", "Network not connected");
        }

    }

    @Override
    public void onStart()
    {
        super.onStart();
        Log.d("instances", "onStart Called, pin = " +pin);
    }

    @Override
    public void onSaveInstanceState(Bundle savedInstanceState)
    {
        super.onSaveInstanceState(savedInstanceState);

        Log.d("instances", "save instance called");

        savedInstanceState.putInt("PIN", pin);
        savedInstanceState.putParcelable(DRINKS_PARCELABLE_NAME, new DrinksParcelable(drinks,mixes));
    }
    @Override
    public void onRestoreInstanceState(Bundle savedInstanceState)
    {
        super.onRestoreInstanceState(savedInstanceState);

        Log.d("instances", "restore instance called");

        pin = savedInstanceState.getInt("PIN");
        DrinksParcelable dp = savedInstanceState.getParcelable(DRINKS_PARCELABLE_NAME);
        drinks = dp.drinks;
        mixes = dp.mixes;
    }




    public void gotoMixView(View view)
    {
        if(net.getStatus() == AsyncTask.Status.RUNNING)
        {
            Toast.makeText(getApplicationContext(), "wait for initialization", Toast.LENGTH_SHORT).show();

        }
        else
        {
            Intent intent = new Intent(this, MakeMixViewActivity.class);
            intent.putExtra(DRINKS_PARCELABLE_NAME, new DrinksParcelable(drinks, mixes));
            startActivity(intent);
        }
    }

    public void gotoSettingsView(View view)
    {
        if(net.getStatus() == AsyncTask.Status.RUNNING)
        {
            Toast.makeText(getApplicationContext(), "wait for initialization", Toast.LENGTH_SHORT).show();

        }
        else
        {
            Intent intent = new Intent(this, PinEntryActivity.class);
            intent.putExtra(DRINKS_PARCELABLE_NAME, new DrinksParcelable(drinks, mixes));
            startActivity(intent);
        }
    }

    public boolean isNetworkConnected()
    {
        ConnectivityManager connMgr = (ConnectivityManager)
                getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isConnected();
    }


    public static Drink[] getMachineDrinks()
    {
        return machine.getDrinks();
    }

    public static void setPin(int pin)
    {
        pinLock.lock();
        MainActivity.pin = pin;
        pinLock.unlock();
    }
    public static int getPin()
    {
        pinLock.lock();
        int ret = pin;
        pinLock.unlock();
        return ret;
    }


}

File: Drink Mixer 9000/app/src/main/java/greenbrier/drinkmixer3000/MainActivity.kt
package greenbrier.drinkmixer3000

import android.content.Context
import android.net.ConnectivityManager
import android.os.AsyncTask
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.LinearLayout

import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView = findViewById<View>(R.id.recyclerView) as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayout.VERTICAL,false)

        val drinks = ArrayList<Drink>() //val is like final in java

        drinks.add(Drink("Cuba Libre", "Coca-Cola, lime, and dark or light rum."))
        drinks.add(Drink("Mojito", "White rum, sugar, lime juice, soda water, and mint."))
        drinks.add(Drink("Long Island Iced Tea", "1/2 ounce triple sec\n" +
                                                                    "1/2 ounce light rum\n" +
                                                                    "1/2 ounce gin\n" +
                                                                    "1/2 ounce vodka\n" +
                                                                    "1/2 ounce tequila\n" +
                                                                    "1-ounce sour mix\n" +
                                                                    "Cola\n" +
                                                                    "Lemon wedge for garnish "))
        drinks.add(Drink("Cosmopolitan","1/2 oz Fresh lime juice, 1 oz Cranberry juice, 1/2 oz Cointreau, 1 1/2 oz Vodka Citron"))
        drinks.add(Drink("Mai Tai", "1 1/2 oz White rum, 1/2 oz Fresh lime juice, 1/2 oz Orange curaçao, 1/2 oz Orgeat syrup, 3/4 oz Dark rum"))

        val adapter = CustomAdapter(drinks)

        recyclerView.adapter = adapter

        if( isNetworkConnected())
        {
            val net = Network()
            net.execute("init")
        }
        else
        {
            Log.d("Network", "Network not connected")
        }

    }

     fun isNetworkConnected(): Boolean
     {
        val connMgr = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connMgr.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnected
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
